sync2jira/upstream_issue.py

```diff
@@ -309,26 +311,26 @@ def github_issues(upstream, config):
         issue['priority'] = ''
         orgname, reponame = upstream.rsplit('/', 1)
         issuenumber = issue['number']
-        default_github_project_fields = config['sync2jira']['default_github_project_fields']
-        project_github_project_fields = config['sync2jira']['map']['github'][upstream]['github_project_fields']
-        github_project_fields = default_github_project_fields | project_github_project_fields
+        github_project_fields = config['sync2jira']['map']['github'][upstream]['github_project_fields']
         variables = {"orgname": orgname, "reponame": reponame, "issuenumber": issuenumber}
-        for fieldname, values in github_project_fields.items():
-            response = requests.post(graphqlurl, headers=headers, json={"query": ghquery, "variables": variables})
-            data = response.json()
-            try:
-                nodes = data['data']['repository']['issue']['projectItems']['nodes'][0]['fieldValues']['nodes']
-                for item in nodes:
-                    if item.get('fieldName') and item.get('fieldName').get('name'):
-                        ghFieldName = item.get('fieldName').get('name')
-                        if next(iter(github_project_fields['priority']['fieldmap'].keys())) == ghFieldName:
-                            issue['priority'] = item.get('name')
-                        if next(iter(github_project_fields['storypoints']['fieldmap'].keys())) == ghFieldName:
-                            issue['storypoints'] = int(item.get('number'))
-            except (TypeError, KeyError) as err:
-                log.debug("Error fetching %s!r from GitHub %s/%s#%s: %s",
-                          ghFieldName, orgname, reponame, issuenumber, err)
-                continue
+        response = requests.post(graphqlurl, headers=headers, json={"query": ghquery, "variables": variables})
+        if response.status_code != 200:
+            log.debug("Error fetching from GitHub: %s", response.text)
+            continue
+        data = response.json()
+        try:
+            nodes = data['data']['repository']['issue']['projectItems']['nodes'][0]['fieldValues']['nodes']
+            for item in nodes:
+                if 'fieldName' in item:
+                    gh_field_name = item.get('fieldName', {}).get('name')
+                    if gh_field_name in github_project_fields['priority']['fieldmap']:
```

Comment:
Given the `if` statement at line 326 and the `get()` with the default value of `{}` in line 325, you don't need the `if` statement at line 324.

On the other hand, if you like and want to keep the if statement at line 324, then you should change line 325 to
```python
                    gh_field_name = item['fieldName'].get('name')
```
That is, if control reaches line 325, then we know that `'fieldName'` is a valid key in `item`, so we don't need to call `get()` to try to get its value -- we can index it unconditionally.

Reply:
Fixed.